server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -1303,6 +1306,31 @@ private static boolean isFiltered(Property[] properties) {
         return properties != null && Arrays.asList(properties).contains(Property.Filtered);
     }
 
+    // RegexValidator
+    public static class RegexValidator implements Writeable, Validator<String> {
+        private Pattern pattern;
+        private String value = "forbidden";
+
+        public RegexValidator(String regex){
+            this.pattern = Pattern.compile(regex);
```

Comment:
This is good.  Compilation is the most performance-sensitive part so it is much better to store the compiled Pattern here rather than the string.